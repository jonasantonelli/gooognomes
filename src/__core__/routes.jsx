import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import Loadable from 'react-loadable';

import Loading from './components/loading/index.jsx';

const GnomeList = Loadable({
    loader: () => import(/* webpackChunkName: "list" */'../gnome/component/container/list/index.jsx'),
    loading: Loading
});

const GnomeDetails = Loadable({
    loader: () => import(/* webpackChunkName: "details" */'../gnome/component/container/details/index.jsx'),
    loading: Loading
});

export const RoutesComponent = ({location}) => (
  <Switch location={location} key="page-switch">
    <Route exact path="/" component={() => <Redirect to={{pathname: "/list"}}/>} />
    <Route exact path="/#" component={() => <Redirect to={{pathname: "/list"}}/>} />
    <Route key="gnomes-list" path="/list" component={GnomeList} />
    <Route key="gnomes-details" path="/details/:id" component={GnomeDetails} />
  </Switch>
);

const Routes = () => (
  <Route
    render={({location}) => <RoutesComponent location={location} />}
  />
);

export default Routes;
