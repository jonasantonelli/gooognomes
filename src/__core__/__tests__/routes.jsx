/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';

import { RoutesComponent } from '../routes.jsx';

describe('Router.jsx component', () => {

  const initialLocation = {
    hash: '',
    pathname: '/list',
    search: '',
    state: undefined
  }

  const wrapper = shallow(<RoutesComponent location={initialLocation} />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
