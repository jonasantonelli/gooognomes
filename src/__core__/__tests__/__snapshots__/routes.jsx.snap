// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router.jsx component should match snapshot 1`] = `
<Switch
  key="page-switch"
  location={
    Object {
      "hash": "",
      "pathname": "/list",
      "search": "",
      "state": undefined,
    }
  }
>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/#"
  />
  <Route
    component={[Function]}
    key="gnomes-list"
    path="/list"
  />
  <Route
    component={[Function]}
    key="gnomes-details"
    path="/details/:id"
  />
</Switch>
`;
